File: samples/GluonSQLiteScala/src/main/scala/com.gluonhq.sqlite/DBUtils.scala
package com.gluonhq.sqlite

import java.io.FileOutputStream
import java.io.IOException
import java.io.InputStream
import java.io.OutputStream
import com.gluonhq.sqlite._

object DBUtils {

    def copyDatabase(pathIni: String, pathEnd: String, name: String): Unit = {

        using(classOf[GluonSQLite].getResourceAsStream(pathIni + name)) { in =>
            using(new FileOutputStream(s"$pathEnd/$name")) { out =>
                val buffer = new Array[Byte](1024)
                Stream.continually(in.read(buffer)).takeWhile(_ != -1).foreach(out.write(buffer, 0, _))
                out.flush()
            }
        }

    }

}

File: samples/GluonSQLite/src/main/java/com/gluonhq/sqlite/DBUtils.java
package com.gluonhq.sqlite;


import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

public class DBUtils {

    public static void copyDatabase(String pathIni, String pathEnd, String name)  {

        try (InputStream myInput = GluonSQLite.class.getResourceAsStream(pathIni+name)) {
            String outFileName =  pathEnd + "/" + name;
            try (OutputStream myOutput = new FileOutputStream(outFileName)) {
                byte[] buffer = new byte[1024];
                int length;
                while ((length = myInput.read(buffer)) > 0) {
                    myOutput.write(buffer, 0, length);
                }
                myOutput.flush();

            } catch (IOException ex) {
                System.out.println("Error " + ex);
            }
        } catch (IOException ex) {
            System.out.println("Error " + ex);
        }
    }
}
